aws/rust-runtime/aws-sigv4/src/http_request/settings.rs

```diff
@@ -61,6 +64,20 @@ pub enum PercentEncodingMode {
     Single,
 }
 
+/// Config value to specify whether URI path normalization should be performed when signing.
+/// <https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html>
+///
+/// URI path normalization is performed based on <https://www.rfc-editor.org/rfc/rfc3986>.
+#[non_exhaustive]
+#[derive(Debug, Eq, PartialEq)]
+pub enum UriPathNormalizationMode {
+    /// We normalize URI paths according to the RFC
+    Enabled,
+
+    /// The exception to this is that we do not normalize URI paths for requests to Amazon S3
```

Comment:
Rephrased so that S3 is one example instead of _the_ example (there may be others we're unaware of right now):
```suggestion
    /// Don't normalize the URI path (S3, for example, rejects normalized paths in some instances)
```

Reply:
Thanks! I like the updated for clarity.